File: trabajos_previos/semana_2/clase_4/herencia_multiple.cpp
#include <iostream>
using namespace std;

// Dos clases
class Mammal {
    public:
        Mammal(){
            cout<< "Mamals can give direct birth." << endl;
        }
};

class WingedAnimal {
    public:
        WingedAnimal(){
            cout << "Winged animal can flap" << endl;
        }
};

// Esta clase tiene de herencoia las otras dos al mismo tiempo
class Bat : public Mammal, public WingedAnimal {};

int main() {
    // El objeto de la tercera clase ejecutará el constructor de las otras dos
    Bat b1;
    return 0;
}

File: trabajos_previos/semana_4/clase_12/herenciaMultiple.py
# Clase mammal
class Mammal:
    def mammal_info(self):
        print("Mammals can give direct birth.")

# Clse WingedAnimal
class WingedAnimal:
    def winged_animal_info(self):
        print("Winged animals can flap.")

# Herenda ambas clases
class Bat(Mammal, WingedAnimal):
    pass

# Objeto de la clase Bat()
b1 = Bat()
# Puede acceder a los métodos de ambas clases
b1.mammal_info()
b1.winged_animal_info()
